modules/apps/27-interchain-accounts/controller/keeper/keeper.go

```diff
@@ -202,32 +212,38 @@ func (k Keeper) GetAllActiveChannels(ctx sdk.Context) []genesistypes.ActiveChann
 }
 
 // SetActiveChannelID stores the active channelID, keyed by the provided connectionID and portID
-func (k Keeper) SetActiveChannelID(ctx sdk.Context, connectionID, portID, channelID string) {
-	store := ctx.KVStore(k.storeKey)
-	store.Set(icatypes.KeyActiveChannel(portID, connectionID), []byte(channelID))
+func (k Keeper) SetActiveChannelID(ctx context.Context, connectionID, portID, channelID string) {
+	store := k.storeService.OpenKVStore(ctx)
+	if err := store.Set(icatypes.KeyActiveChannel(portID, connectionID), []byte(channelID)); err != nil {
+		panic(err)
+	}
 }
 
 // IsActiveChannel returns true if there exists an active channel for the provided connectionID and portID, otherwise false
-func (k Keeper) IsActiveChannel(ctx sdk.Context, connectionID, portID string) bool {
+func (k Keeper) IsActiveChannel(ctx context.Context, connectionID, portID string) bool {
 	_, ok := k.GetActiveChannelID(ctx, connectionID, portID)
 	return ok
 }
 
 // GetInterchainAccountAddress retrieves the InterchainAccount address from the store associated with the provided connectionID and portID
-func (k Keeper) GetInterchainAccountAddress(ctx sdk.Context, connectionID, portID string) (string, bool) {
-	store := ctx.KVStore(k.storeKey)
+func (k Keeper) GetInterchainAccountAddress(ctx context.Context, connectionID, portID string) (string, bool) {
+	store := k.storeService.OpenKVStore(ctx)
 	key := icatypes.KeyOwnerAccount(portID, connectionID)
 
-	if !store.Has(key) {
+	bz, err := store.Get(key)
+	if len(bz) == 0 {
 		return "", false
 	}
+	if err != nil {
+		panic(err)
+	}
 
-	return string(store.Get(key)), true
+	return string(bz), true // todo: why the cast?
```

Comment:
address value is stored as string... similar reasoning to the ics29 question you had on counterparty address, could store address bytes but string is more human readable